homeassistant/components/blebox/cover.py

```diff
@@ -0,0 +1,109 @@
+"""BleBox cover entity."""
+
+import logging
+
+from homeassistant.components.cover import (
+    ATTR_POSITION,
+    STATE_CLOSED,
+    STATE_CLOSING,
+    STATE_OPENING,
+    SUPPORT_CLOSE,
+    SUPPORT_OPEN,
+    SUPPORT_SET_POSITION,
+    SUPPORT_STOP,
+    CoverDevice,
+)
+
+from . import BleBoxEntity, async_create_blebox_entities
+from .const import (
+    BLEBOX_TO_HASS_COVER_STATES,
+    BLEBOX_TO_HASS_DEVICE_CLASSES,
+    DOMAIN,
+    PRODUCT,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add):
+    """Set up a BleBox entry."""
+
+    product = hass.data[DOMAIN][config_entry.entry_id][PRODUCT]
+    return await async_create_blebox_entities(
+        product, async_add, BleBoxCoverEntity, "covers"
+    )
+
+
+class BleBoxCoverEntity(BleBoxEntity, CoverDevice):
+    """Representation of a BleBox cover feature."""
+
+    @property
+    def state(self):
+        """Return the equivalent HA cover state."""
+        return BLEBOX_TO_HASS_COVER_STATES[self._feature.state]
+
+    @property
+    def device_class(self):
+        """Return the device class."""
+        return BLEBOX_TO_HASS_DEVICE_CLASSES[self._feature.device_class]
+
+    @property
+    def supported_features(self):
+        """Return the supported cover features."""
+        position = SUPPORT_SET_POSITION if self._feature.is_slider else 0
+        stop = SUPPORT_STOP if self._feature.has_stop else 0
+
+        return position | stop | SUPPORT_OPEN | SUPPORT_CLOSE
+
+    @property
+    def current_cover_position(self):
+        """Return the current cover position."""
+        position = self._feature.current
+        if position == -1:  # possible for shutterBox
+            name = self.name
+            _LOGGER.warning(
+                "Position for %s is unknown. Try calibrating the device.", name
+            )
```

Comment:
We don't want side effects in entity state properties. Please remove the logging.
```suggestion
```